test-dev/compare_mixer_data.c

```diff
@@ -80,6 +80,9 @@ static void _compare_mixer_data(const char *mod, const char *data, int loops, in
 			}
 			fail_unless(abs(vi->pos0 - pos0) <= 1, "position mismatch");
 			if (num >= 11) {
+				if (cutoff >= 254 && vi->filter.cutoff >= 254) {
+					cutoff = vi->filter.cutoff;
```

Comment:
How do we hit this?

Reply:
Cutoff values of 254 and 255 are interchangeable due to libxmp doubling channel cutoff values. The maximum channel filter cutoff value is 127 in IT, which libxmp doubles to 254, but libxmp also uses 255 elsewhere. Most of the mixer tests contain cutoff values of 255 due to this, but this patch causes some of them to change to 254, since the cutoff isn't being reset to 255 anymore. This check is to suppress a bunch of mixer tests that broke from this minor and mostly irrelevant change (I think it was a couple dozen?).

The doubling is presumably to appease IMAGO Orpheus modules, which have a cutoff effect that *maybe* uses its full parameter range. I haven't figured out how to get the filter to work in IMAGO Orpheus with DOSBox yet so I can't reasonably verify this. This doubling behavior should probably die (and the player should not divide the cutoff value by the filter envelope range), but I can't do much about it without being able to test that format. After that gets fixed the mixer data tests can probably just normalize the new cutoff values to 255 and this check can probably be removed.

Reply:
OK.

Reply:
> The doubling is presumably to appease IMAGO Orpheus modules, which have a cutoff effect that maybe uses its full parameter range

I think it's more for the sake of increased precision after multiplying the filter envelope with the cutoff. True, the cutoff itself is limited to 0...127, but by using the filter envelope you could get fractional values. IT itself uses even higher precision here in fact (127*255 instead of just 127*2 possible values).